CI/integration_tests/calculators/test_angular_distribution_function.py

```diff
@@ -69,3 +70,21 @@ def test_project(traj_file, true_values, tmp_path):
         computation["NaCl"].data_dict[item].pop("max_peak")
 
     assertDeepAlmostEqual(computation["NaCl"].data_dict, true_values, decimal=-2)
+
+
+def test_low_memory(traj_file, true_values, tmp_path):
+    """Test the ADF called from the project class"""
+    mds.config.memory_fraction = compute_memory_fraction(0.1)
+    os.chdir(tmp_path)
+    project = mds.Project()
+    project.add_experiment(
+        "NaCl", simulation_data=traj_file, timestep=0.002, temperature=1400
+    )
+
+    computation = project.run.AngularDistributionFunction(plot=False)
+
+    for item in computation["NaCl"].data_dict:
+        computation["NaCl"].data_dict[item].pop("max_peak")
+
+    assertDeepAlmostEqual(computation["NaCl"].data_dict, true_values, decimal=-2)
+    mds.config.memory_fraction = 0.5
```

Comment:
I don't like the code duplication of changing the memory fraction back. Could we use a helpfer function for all tests like:

```python
@contextlib.contextmanager
def reduce_memory_fraction(memory_fraction: float = 0.1):
  default = mds.config.memory_fraction
  mds.config.memory_fraction = compute_memory_fraction(memory_fraction)

try:
  yield
finally:
  mds.config.memory_fraction = default
```

Reply:
This also reduced code duplication because we can write:

```python
@pytest.mark.parametrize("memory_fraction": (mds.config.memotry_fraction, 0.1))
def my_test(memory_fraction):
   with reduce_memory_fraction(memory_fraction):
      ...
```

Reply:
This looks awesome, I would much prefer it be like this, could you do some of these so I see what you mean as I don't fully see how to implement that and have it run through all the tests but something like this is what I hoped I could do but I went for the simple way so I could actually test the calculators which uncovered some faults.

Reply:
That's awesome thanks!